lib/payrix/resource/base.rb

```diff
@@ -207,7 +207,7 @@ def build_search(values = {})
       def validate_response
         if @response.has_errors?
           if Payrix.configuration.exception_enabled
-            raise Payrix::Exceptions::ApiError.new('There are errors in the response')
+            raise Payrix::Exceptions::ApiError.new("There are errors in the response, #{@response.errors}")
```

Comment:
This error class inherits from `Payrix::Exceptions::Base`, which inherits from `RuntimeError`, hence it can only take one param, so I concatenated the errors payload and that would raise an error in this format 👇.

```ruby
Payrix::Exceptions::ApiError:
       There are errors in the response, [{"code"=>13, "severity"=>4, "msg"=>"No such action and/or resource", "errorCode"=>"invalid_resource"}]

``` 

I also thought of allowing `Payrix::Exceptions::ApiError` to accept two params `message` and `errors`, although since it's only used here (in `Payrix::Resource::Base`), I decided to concatenate it directly here, let me know what you think.

```ruby
# that would make api_error.rb be...
module Payrix
  module Exceptions
    class ApiError < Base
      def initialize(message, errors=nil)
        super(errors.nil? ? message : "#{message}, #{errors}")
      end
    end
  end
end

# in resource/base.rb, we would have...
...
raise Payrix::Exceptions::ApiError.new('There are errors in the response', @response.errors)
...
```

Reply:
I like the approach you decided to go with.

Both approaches you identified _would_ solve the issue. The latter approach though is more heavy-handed in my opinion, since it redefines the interface for _all_ errors. I think it's a good idea to make these sorts of changes on an isolated level and observe the need for generalizing it later on.